cli/mender-artifact/main.go

```diff
@@ -94,13 +94,132 @@ func run() error {
 			Usage: "Full path to the state script(s). You can specify multiple " +
 				"scripts providing this parameter multiple times.",
 		},
+		/////////////////////////
+		// Version 3 specifics.//
+		/////////////////////////
+		cli.StringSliceFlag{
+			Name:  "artifact-name-depends",
+			Usage: "Sets the name(s) of the artifact(s) which this update depends upon",
+		},
+		cli.StringFlag{
+			Name:  "provides-group",
+			Usage: "The group the artifact provides",
+		},
+		cli.StringSliceFlag{
+			Name:  "depends-groups",
+			Usage: "The group(s) the artifact depends on",
+		},
+		cli.StringFlag{
+			Name:  "depends-rootfs-image-checksum",
+			Usage: "The checksum of the rootfs image which this artifact depends upon",
+		},
+		cli.StringFlag{
+			Name:  "provides-rootfs-image-checksum",
+			Usage: "The checksum of the rootfs image which this artifact provides",
+		},
+	}
+
+	//
+	// Update modules: module-image
+	//
+	writeModuleCommand := cli.Command{
+		Name:   "module-image",
+		Action: writeModuleImage,
+		Usage:  "Writes Mender artifact for an update module",
+		UsageText: "Writes a generic Mender artifact that will be used by an update module. " +
+			"This command is not meant to be used directly, but should rather be wrapped by an " +
+			"update module build command, which prepares all the necessary files and headers " +
+			"for that update module.",
+	}
+
+	writeModuleCommand.Flags = []cli.Flag{
+		cli.StringSliceFlag{
+			Name: "device-type, t",
+			Usage: "Type of device(s) supported by the update. You can specify multiple " +
+				"compatible devices providing this parameter multiple times.",
```

Comment:
Most of them, but not all. The `--update` / `-u` flag, although it has an equivalent in the module image, isn't really the same thing thing, because there can be many files, with different roles, depending on how the update module works. It can be a header, a manifest, a signature, a package, anything really. So I opted to call this `--file` / `-f` instead.

For those flags that matched, I copied over all the short forms.